src/Authentication/Authentication/Cmdlets/InvokeGraphRequest.cs

```diff
@@ -0,0 +1,965 @@
+            var httpException = new HttpResponseException(errorMessage, currentResponse);
+            var errorRecord = new ErrorRecord(httpException, "WebCmdletWebResponseException", ErrorCategory.InvalidOperation, httpRequestMessage);
+            var detailMsg = httpResponseMessageFormatter.ReadAsStringAsync()
+                .GetAwaiter()
+                .GetResult();
+            if (!string.IsNullOrEmpty(detailMsg))
+            {
+                errorRecord.ErrorDetails = new ErrorDetails(detailMsg);
+            }
+
+            return errorRecord;
+        }
+
+        private void ReportRequestStatus(HttpRequestMessage requestMessage)
+        {
+            long requestContentLength = 0;
+            if (requestMessage.Content != null)
+            {
+                requestContentLength = requestMessage.Content.Headers.ContentLength.Value;
+            }
+            var reqVerboseMsg = StringFormatCurrentCulture("{0} {1} with {2}-byte payload",
+                requestMessage.Method,
+                requestMessage.RequestUri,
+                requestContentLength);
+            WriteVerbose(reqVerboseMsg);
+        }
+
+        private void ReportResponseStatus(HttpResponseMessage responseMessage)
+        {
+            var contentType = ContentHelper.GetContentType(responseMessage);
+            var respVerboseMsg = StringFormatCurrentCulture("received {0}-byte response of content type {1}",
+                responseMessage.Content.Headers.ContentLength,
+                contentType);
+            WriteVerbose(respVerboseMsg);
+        }
+
+        private static string StringFormatCurrentCulture(string format, params object[] args)
+        {
+            return string.Format(CultureInfo.CurrentCulture, format, args);
+        }
+
+        private static string FormatDictionary(IDictionary content)
+        {
+            if (content == null)
+                throw new ArgumentNullException(nameof(content));
+
+            var bodyBuilder = new StringBuilder();
+            foreach (string key in content.Keys)
+            {
+                if (0 < bodyBuilder.Length) bodyBuilder.Append("&");
+
+                var value = content[key];
+
+                // URLEncode the key and value
+                var encodedKey = WebUtility.UrlEncode(key);
+                var encodedValue = string.Empty;
+                if (value != null) encodedValue = WebUtility.UrlEncode(value.ToString());
+
+                bodyBuilder.AppendFormat("{0}={1}", encodedKey, encodedValue);
+            }
+
+            return bodyBuilder.ToString();
+        }
+
+        private HttpRequestMessage GetRequest(Uri uri)
+        {
+            var requestUri = PrepareUri(uri);
+            var httpMethod = GetHttpMethod(Method);
+            // create the base WebRequest object
+            var request = new HttpRequestMessage(httpMethod, requestUri);
```

Comment:
`using var` ?

Reply:
C# 8.0 feature simplifying using statements. 
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-statement

Reply:
We should not be using C# 8 in this project. Some C# 8 features depend on .NET Standard 2.1 and .NET Core 3.x, see https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version.

To maintain consistency with the generated PS module projects, we should always use C# 7.1.

Reply:
I saw `using var` being used on other files, so I assumed that we can use C# 8. @finsharp, per @peombwa's comment, please revert them in `InvokeGraphRequest.cs` as well.

`IDisposable` objects should dispose properly. Even if you can't have `using var`, you can always have `using` blocks.

Reply:
Will revert to 7.1